draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -335,20 +332,84 @@ hop(AS(i), AS(j)) =  / Else, "Provider+" if the VAP-SPAS entry
         In the current document, routes with AS_SET are given Invalid evaluation in the AS_PATH verification procedures (<xref target="Upflow"/> and <xref target="Downflow"/>).
         See <xref target="mitigation"/> for how routes with Invalid AS_PATH are handled.
       </t>
+
+    <section title="Semantics of Valid, Invalid, and Unknown" anchor="semantics">
+      <t>
+        The ASPA-based path verification algorithms (<xref target="Upflow"/> and <xref target="Downflow"/>) evaluate an AS_PATH as Valid, Invalid, or Unknown.
+        'Invalid' outcome means that the AS_PATH contains a route leak (i.e., valley-free violation) based on the available ASPAs.
+        Even if the actual attack on the AS_PATH attribute was a forged-origin hijack or forged-path-segment hijack
+        or other AS path modification,
+        and the associated conditions permit its detectability by ASPA (see <xref target="property"/>), it would be effectively manifest as a route leak and hence ASPA would detect the route as Invalid.
+        The algorithms do not further identify the specific type of anomaly.
+        'Valid' outcome means that sufficient ASPA information is available to determine that the AS_PATH attribute as presented is route-leak free (i.e., valley free). When the outcome is not Valid or Invalid, then it is Unknown by default.
+      </t>
+    </section>
+    <section title="Basic Principles of Route Leak Detection and Reason for Separate Upstream and Downstream Algorithms" anchor="basic">
+      <t>
+        In <xref target="Upflow"/> and <xref target="Downflow"/> below, the terms "upstream path" and "downstream path" generally refer to AS paths received in the upstream direction (from a customer or a lateral peer) and the downstream direction (from a provider), respectively.
```

Comment:
Thanks for the catch. They are just typos. 
s/ received in the upstream direction (from a customer or a lateral peer) and the downstream direction (from a provider) / received in the 'up' direction (i.e., from a customer or a lateral peer) and the 'down' direction (i.e., from a provider) /

Reply:
While better I would prefer something where `upflow` is not explained with the term `up`.
The problem is that `up` and `down` make sense in e.g. a building but it is far less clear in this context.
My other issue is that roles are also depending on the view point. When I configure a neighbor on a router do I specify his role towards me or is it my role towards him. This little difference suddenly requires to rethink `up` and `down` since the roles are swapped.

I know this is a tough nut to crack but it is one of the main issues I had to fight with during implementation.
Mapping `up`, `down`, `first`, `last`, `customer`, `provider`, ... to my environment and therefor my mental image.
This is why I requested to write and number the as sequences in a form that people expect.
Every time a direction term is used it adds load to the mental image people need to create.

Reply:
>While better I would prefer something where upflow is not explained with the term up.

OK. I get it now. I will make the following substitution:
s/The term "upstream path" generally refers to AS paths received in the
   upstream direction (from a customer, lateral peer, RS-client, RS) and
   "downstream path" refers to AS paths received in the downstream
   direction (from a provider or mutual-transit neighbor)./
   The term "upstream path" refers to AS paths received 
   from a customer, lateral peer, RS-client, or RS, and
   "downstream path" refers to AS paths received
   from a provider or mutual-transit neighbor.

Reply:
This is better. Will check in the new draft and comment there.